<launch>
    <!-- LiDAR odom, disable when other visual odometry devices are present. -->
    <!--node pkg="rf2o_laser_odometry" type="rf2o_laser_odometry_node" name="rf2o_laser_odometry" output="screen">
        <param name="odom_frame_id" value="vo"/>
        <param name="base_frame_id" value="base_link"/>
        
        <param name="laser_scan_topic" value="/scan"/>
        <param name="odom_topic" value="/vo"/>
        
        <param name="publish_tf" value="false"/>
        <param name="init_pose_from_topic" value=""/>
        <param name="verbose" value="false" />
        <param name="freq" value="20.0"/>
    </node-->
    
    <!--node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
        <param name="base_frame" value="base_link" />
        <param name="map_frame" value="fake_odom" />
        <param name="odom_frame" value="odom"/>

        <remap from="map" to="fake_map"/>
        
        <param name="map_size" value="1024"/>
        <param name="map_resolution" value="0.05"/>

        <param name="scan_subscriber_queue_size" value="2"/>
        <param name="advertise_map_service" value="false"/>

        <param name="use_tf_scan_transformation" value="true"/>
        <param name="use_tf_pose_start_estimate" value="false"/>
        
        <param name="pub_map_odom_transform" value="false"/> 
        <param name="pub_map_scanmatch_transform" value="false"/>
        <param name="pub_odometry" value="true" />

        <param name="tf_map_scanmatch_transform_frame_name" value="odom" />
    </node-->
    
    <!-- IMU odom -->
    <!-- imu_data vel, acc -->
    <!--node pkg="map_follower" type="imuBroadcast.py" name="imu_broadcaster" output="screen"/-->

    <!-- <node pkg="laser_scan_matcher" type="laser_scan_matcher_node" name="laser_scan_matcher" output="screen">
        <param name="use_odom" value="false"/>
        <param name="use_imu" value="true"/>
        <remap from="imu/data" to="imu_data"/>
        <param name="publish_tf" value="true"/>
        <param name="fixed_frame" value="odom"/>
    </node> -->

    <!-- Mixer -->
    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
        <param name="odom_used" value="true"/>
        <remap from="/imu" to="/cam_t1/imu"/>
        <param name="imu_used" value="true"/>
        <remap from="/vo" to="/cam_t1/odom/sample"/>
        <param name="vo_used" value="true"/>

        <param name="sensor_timeout" value="1.0"/>
        <param name="debug" value="false"/>
        <param name="self_diagnose" value="false"/>
        
        <param name="output_frame" value="odom"/>
        <param name="base_footprint_frame" value="base_link"/>
        <param name="freq" value="20.0"/>
    </node>
    
</launch>
